resin/models/application.py

```diff
@@ -311,3 +313,152 @@ def disable_rolling_updates(self, app_id):
             'application', 'PATCH', params=params, data=data,
             endpoint=self.settings.get('pine_endpoint')
         )
+
+    def enable_device_urls(self, app_id):
+        """
+        Enable device urls for all devices that belong to an application
+
+        Args:
+            app_id (str): application id.
+
+        Returns:
+            OK/error.
+
+        Examples:
+            >> > resin.models.application.enable_device_urls('5685')
+            'OK'
+
+        """
+
+        params = {
+            'filter': 'belongs_to__application',
+            'eq': app_id
+        }
+        data = {
+            'is_web_accessible': True
+        }
+
+        return self.base_request.request(
+            'device', 'PATCH', params=params, data=data,
+            endpoint=self.settings.get('pine_endpoint')
+        )
+
+    def disable_device_urls(self, app_id):
+        """
+        Disable device urls for all devices that belong to an application.
+
+        Args:
+            app_id (str): application id.
+
+        Returns:
+            OK/error.
+
+        Examples:
+            >> > resin.models.application.disable_device_urls('5685')
+            'OK'
+
+        """
+
+        params = {
+            'filter': 'belongs_to__application',
+            'eq': app_id
+        }
+        data = {
+            'is_web_accessible': False
+        }
+
+        return self.base_request.request(
+            'device', 'PATCH', params=params, data=data,
+            endpoint=self.settings.get('pine_endpoint')
+        )
+
+    def grant_support_access(self, app_id, valid_period):
+        """
+        Grant support access to an application until a specified time.
+
+        Args:
+            app_id (str): application id.
+            valid_period (int): valid period in hour.
```

Comment:
@pimterry, I select hours as an unit since we use it in our dashboard and IMO, timestamp is a bit harder for end-users to use. Any reason why we don't use the unit that we use on the dashboard? Anyway, I'm fine with timestamp as well so it's up to you.

Reply:
Would setting the parameter as a timestamp (instead of as a number of hours) mean that the user has to deal with timezone conversions?

Reply:
> Would setting the parameter as a timestamp (instead of as a number of hours) mean that the user has to deal with timezone conversions?

Should be fine. In JS at least it's pretty easy to do timezone agnostically: you can use either `Date.now() + X` or `Date.parse('2017-11-29T14:40:00')`, and you'll get sensible results.

> I select hours as an unit since we use it in our dashboard and IMO, timestamp is a bit harder for end-users to use. Any reason why we don't use the unit that we use on the dashboard? Anyway, I'm fine with timestamp so the final call is up to you.

Imo it's simpler this way as it's a closer representation of what the API is really doing, while still being easy to map into any UI representation you like (including relative times). This means that the API requests are closer to the SDK methods, and you can do things like set the timestamp, and then read it from `is_accessible_by_support_until__date`, and see the same value.

It's just a bit simpler and more predictable than relative time, and it's easy to convert timestamps to relative time if you want to. If there's no strong reasons not to, I'd rather stick with timestamps.

Reply:
@pimterry, makes sense. I've changed to timestamp. Can you check again?